src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -76,7 +76,7 @@ public int getMinWeightMagnitude() {
         return MIN_WEIGHT_MAGNITUDE;
     }
 
-    private static boolean hasInvalidTimestamp(TransactionViewModel transactionViewModel) {
+    private boolean hasInvalidTimestamp(TransactionViewModel transactionViewModel) {
```

Comment:
I agree with removing `static` on all `public` methods because you don't want to have a global scope `TransactionValidator` logic. However on *private* methods I do prefer to keep the `static` modifier.
  
This is because of 2 reasons:
1. I can quickly know that the method doesn't change the instance state by just looking at the signature.
2. There may be better optimizations that the JIT will do now that the method is marked as static. Today modern JVM are heavily optimized and I am not sure how important this consideration is nowadays. Still the results will be JVM dependent.

Reply:
Atm we use this method to filter old transactions that are still being gossiped about in the network. Since the cut off time is the timestamp of the global snapshot that is not really a big issue in the current environment since it is quite unlikely that somebody has a wrong time set in his node that dates back before the global snapshot.

When we introduce local snapshots and consequently local cutoff times it can however happen (and it does), that we face a transaction with an invalid timestamp that dates back before our cutoff time, even tho it was issued after.

To still be able to consider this transaction valid if we need it for solidification we have to check if the transaction was part of the TransactionRequester. And to be able to access the transactionRequester instance - this method has to be non static.

Regarding 2: I don't think that these kind of micro-optimizations play any role at all - especially if they help us solve a problem that otherwise would break the synced state of a node.

see: https://github.com/iotadevelopment/iri/commit/40df07b8fbf8a7fc4020e4ffc4d700b07b081f97

Reply:
:+1:
If you need access to the object than there is no choice. 
I still like making private methods static if they don't need the instance.

Reply:
Even though I am fine with this change, note that the fact that the `TransactionValidator` has a `TransactionRequester` instance is a violation of the Single Responsibility Principle, and in my opinion it shouldn't be there. `TransactionValidator` should ultimately become a stateless Service.

Of course that it is not your fault that IRI is this way and definitely you should be more concerned about getting local snapshots done rather than fixing IRI architecture.
I just wanted to merge now and I looked at the code again and had to share this thought.